src/filesystem.cpp

```diff
@@ -1733,6 +1733,27 @@ utils::optional<std::string> get_independent_binary_file_path(const std::string&
 	return full_path.generic_string();
 }
 
+utils::optional<std::string> get_wml_binary_file_path(const std::string& type, const std::string& filename)
```

Comment:
I'm kind of wondering why this is needed. It seems like the goal is to resolve the path using binary path rules and then unresolve it using WML rules? And then you're going to just pass it to `get_wml_location` to resolve it again. Why do you need to resolve the path twice?

Reply:
i need to pass the wml path when invoking a lus script. (in particuar lua needs the wml path so that things like `wesnoth.("./sdfg.lua"))` are able to resolve the current directory correctly.

Reply:
Okay, I guess that makes sense, though it's still a bit weird.